frontend/src/App.tsx

```diff
@@ -418,6 +419,18 @@ class App extends PureComponent<Props, State> {
         }
       )
 
+      if (this.elementListBuffer) {
+        const ids = this.elementListBuffer.main
+          .map(e =>
+            (e as ImmutableMap<string, any>)
+              .get((e as ImmutableMap<string, any>).get("type"))
+              .get("id")
+          )
+          .filter(x => x != null)
+          .toArray()
+        this.widgetMgr.clean(ids as [string])
+      }
+
```

Comment:
The idea here is to get all widget that are shown on the page. Need to refactor and include sidebar.